MNE/disp3D/3DObjects/brain/brainsurfacesettreeitem.cpp

```diff
@@ -243,6 +250,70 @@ bool BrainSurfaceSetTreeItem::addData(const MNESourceSpace& tSourceSpace, Qt3DCo
 
 //*************************************************************************************************************
 
+BrainRTSourceLocDataTreeItem* BrainSurfaceSetTreeItem::addData(const MNESourceEstimate& tSourceEstimate, const MNEForwardSolution& tForwardSolution)
+{
+    if(!tSourceEstimate.isEmpty()) {
+        //Add source estimation data as child
+        if(this->findChildren(Data3DTreeModelItemTypes::RTSourceLocDataItem).size() == 0) {
+            //If rt data item does not exists yet, create it here!
+            if(!tForwardSolution.isEmpty()) {
+                m_pBrainRTSourceLocDataTreeItem = new BrainRTSourceLocDataTreeItem();
+
+                QList<QStandardItem*> list;
+                list << m_pBrainRTSourceLocDataTreeItem;
+                list << new QStandardItem(m_pBrainRTSourceLocDataTreeItem->toolTip());
+                this->appendRow(list);
+
+                connect(m_pBrainRTSourceLocDataTreeItem, &BrainRTSourceLocDataTreeItem::rtVertColorChanged,
+                        this, &BrainSurfaceSetTreeItem::onRtVertColorChanged);
+
+                //Divide into left right hemi
+                QList<QStandardItem*> itemList = this->findChildren(Data3DTreeModelItemTypes::HemisphereItem);
+
+                BrainSurfaceTreeItem* pSurfaceTreeItemLeft = new BrainSurfaceTreeItem();
+                BrainSurfaceTreeItem* pSurfaceTreeItemRight = new BrainSurfaceTreeItem();
+                BrainAnnotationTreeItem* pAnnotTreeItemLeft = new BrainAnnotationTreeItem();
+                BrainAnnotationTreeItem* pAnnotTreeItemRight = new BrainAnnotationTreeItem();
+
+                for(int j = 0; j < itemList.size(); j++) {
+                    if(BrainHemisphereTreeItem* pHemiItem = dynamic_cast<BrainHemisphereTreeItem*>(itemList.at(j))) {
+                        if(pHemiItem->data(Data3DTreeModelItemRoles::SurfaceHemi).toInt() == 0) {
+                            pSurfaceTreeItemLeft = pHemiItem->getSurfaceItem();
+                            pAnnotTreeItemLeft = pHemiItem->getAnnotItem();
+                        } else if(pHemiItem->data(Data3DTreeModelItemRoles::SurfaceHemi).toInt() == 1) {
+                            pSurfaceTreeItemRight = pHemiItem->getSurfaceItem();
+                            pAnnotTreeItemRight = pHemiItem->getAnnotItem();
+                        }
+                    }
+                }
+
+                m_pBrainRTSourceLocDataTreeItem->init(tForwardSolution,
```

Comment:
call init function only when pSurfaceTreeItemLeft, ..Right, pAnnotTreeItemLeft, ...Right, etc is pointing to something, i.e. is not Q_NULLPTR anymore so guard this function with if(pSurfaceTreeItemLeft && pAnnotTreeItemLeft &&..) {  init }